lib/livebook/notebook_manager.ex

```diff
@@ -256,7 +256,16 @@ defmodule Livebook.NotebookManager do
     end
   end
 
-  defp load_file(%{file_system_id: file_system_id, file_system_type: file_system_type, path: path}) do
+  defp load_file(%{file_system_id: file_system_id, path: path} = file) do
+    # In the past, not all files had a file_system_type, so we need to detect one from the id.
+    file_system_type =
+      Map.get_lazy(file, :file_system_type, fn ->
+        case file_system_id do
+          "local" -> "local"
+          "s3-" <> _ -> "s3"
+        end
+      end)
```

Comment:
Now that we run migrations before NotebookManager, we can make this a migration! I can can do it in a separate PR :)

Reply:
It is your call. This code is harmless enough as is. :)